src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipe.cs

```diff
@@ -189,7 +199,7 @@ internal static class EventPipeInternal
         // These PInvokes are used by the configuration APIs to interact with EventPipe.
         //
         [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
-        internal static extern UInt64 Enable(string outputFile, uint circularBufferSizeInMB, long profilerSamplingRateInNanoseconds, EventPipeProviderConfiguration[] providers, int numProviders);
+        internal static extern UInt64 Enable(string outputFile, uint circularBufferSizeInMB, long profilerSamplingRateInNanoseconds, EventPipeProviderConfiguration[] providers, int numProviders, bool enableAllEvents);
```

Comment:
I would really rather not have the 'All Events' special case wired into the 'Enable' API.   It is just not that valuable.   I was hoping instead that we could just treat '*' as a provider, and then have a special case that interprets it as a FILTER (set) rather than a particular provider (This also allows you extend this to general pattern matching (e.g Runtime*).